common/src/main/scala/com/gu/multimedia/storagetier/framework/MessageProcessingFramework.scala

```diff
@@ -186,6 +187,49 @@ class MessageProcessingFramework (ingest_queue_name:String,
     }
   }
 
+  def withChannel[T](block:Channel=>Try[T]):Try[T] = {
+    Try { conn.createChannel() }.flatMap(ch=>{
+      block(ch) match {
+        case result@Success(_)=>result
+        case err@Failure(_)=>
+          Try { channel.close() } //swallow the exception if it fails
+          err
+      }
+    })
+  }
+
+
+  def bulkSendMessages[T:io.circe.Encoder](outputExchange:String, routingKey:String, msgs:Seq[T], retry:Int=0) = {
+    withChannel( {chan =>
+      for {
+        _ <- Try { chan.exchangeDeclare(outputExchange, "topic", true)}
+        result <- {val sendResults = msgs.map(_.asJson.noSpaces)
+        .map(stringContent => internalSendMsg(chan, outputExchange, routingKey, stringContent))
+        val failures = sendResults.collect({ case Failure(err) => err })
+          if(failures.nonEmpty){
+            logger.error(s"${failures.length} messages failed to send to $outputExchange.")
+            failures.foreach(err => {
+              logger.error(s"\t${err.getMessage}")
+              logger.debug(s"${err.getMessage}", err)
+            })
+            Failure(new RuntimeException(s"${failures.length} / ${msgs.length} messages failed to send to $outputExchange as $routingKey"))
+          } else {
+            Success(())
+          }
+        }
+      } yield result
+    })
+  }
+  private def internalSendMsg(chan:Channel, outputExchange:String, routingKey:String, stringContent:String) = Try {
+
+    val msgProps = new AMQP.BasicProperties.Builder()
+      .contentType("application/json")
+      .contentEncoding("UTF-8")
+      .messageId(UUID.randomUUID().toString)
+      .build()
+
+    chan.basicPublish(outputExchange, routingKey, false,msgProps, stringContent.getBytes(cs) )
```

Comment:
I am guessing `cs` is the character set, but I can't see where it's defined?

Reply:
Its declared at line 45